### PR TITLE
Make space for documentation aside during followup completion select

### PR DESCRIPTION
**Pull request by mgsloan**
The goal of #7115 appears to be to limit the disruptiveness of completion documentation load causing the completion selector to move around. The approach was to debounce load of documentation via a setting `completion_documentation_secondary_query_debounce`. This particularly had a nonideal interaction with #21286, where now this debounce interval was used between the documentation fetches of every individual completion item.

I think a better solution is to continue making space for documentation to be shown as soon as any documentation is shown. #21704 implemented part of this, but it did not persist across followup completions.

Release Notes:

- Fixed completion list moving around on load of documentation. The previous approach to mitigating this was to rate-limit the fetch of docs, configured by a `completion_documentation_secondary_query_debounce` setting, which is now deprecated.

### CODE REVIEW

File: crates/editor/src/editor.rs [hunk 1]
```diff
@@ -1007,7 +1005,7 @@ struct CompletionsMenu {
     matches: Arc<[StringMatch]>,
     selected_item: usize,
     scroll_handle: UniformListScrollHandle,
-    selected_completion_resolve_debounce: Option<Arc<Mutex<DebouncedDelay>>>,
+    enable_completion_resolution: bool,
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor.rs:1008]:
```suggestion
    resolve_completions: bool,
```

NIT, some naming proposals.
Also, wondering: we disable this for assistant completions only, those aren't supposed to have any resolve provided anyway, are they? If true, we can remove the field entirely as it's not necessary.

File: crates/editor/src/editor_settings.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ pub struct EditorSettings {
     pub hover_popover_enabled: bool,
     pub show_completions_on_input: bool,
     pub show_completion_documentation: bool,
-    pub completion_documentation_secondary_query_debounce: u64,
+    pub completion_documentation_secondary_query_debounce: Option<u64>,
```
Comment (SomeoneToIgnore) [inline crates/editor/src/editor_settings.rs:15]: NIT: I think we can silently ignore that and remove entirely, as we did with a few other deprecated items, stating that in the release notes.

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you for fixing all this.

File: crates/editor/src/editor_settings.rs (hunk 2 above)
Comment (mgsloan) [reply crates/editor/src/editor_settings.rs:15]: Well that explains why I don't see any other deprecated settings!  Thanks

File: crates/editor/src/editor.rs (hunk 1 above)
Comment (mgsloan) [reply crates/editor/src/editor.rs:1008]: Good point, changed the name!

**Pull request merged**